### ISSUE TITLE
feat(speculative): Muse Glimmer DFlash assistant drafter (muse_glimmer_assistant, 5 sliding-attention layers, bidirectional sliding mask) on the existing DFlash round loop

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1343]
## Summary

`meta-models/Muse-Glimmer-30B-assistant` is a DFlash-style block-parallel drafter for `meta-models/Muse-Glimmer-30B` (`model_type: muse_glimmer_assistant`): five sliding-attention decoder layers with per-head q/k norms, fed with five of the target's residual streams through an `encoder.fc` projection, predicting a 16-token block in one non-causal forward under a bidirectional sliding-window mask, and borrowing the target's raw embedding table and untied `lm_head`. mlxcel serves Muse Glimmer (`src/models/muse_glimmer*.rs`) and has the DFlash round loop, but `mlxcel-server` rejects every `--draft-*` flag on a Muse target and the existing drafter model is the Qwen-shaped one (no q/k norms, no mask, plain growing cache, different key names). This change adds the assistant model, the Muse target adapter (hidden capture plus rotating-cache rollback), and the dispatch wiring so `--model-draft <assistant-dir>` runs the DFlash round loop on Muse Glimmer text requests with the exact-match verify rule.

## Current behavior

- `src/server/startup.rs:787-802` (`validate_muse_glimmer_unsupported_startup`): `ensure!(startup.draft_model_path.is_none() && startup.draft_kind.is_none() && startup.draft_block_size.is_none(), "Muse Glimmer VLM does not support speculative decoding or DFlash; ...")`, pinned by `src/server/muse_glimmer_startup_guard_tests.rs:86-97`. `docs/supported-models.md:562`: "Muse Glimmer is not a speculative or DFlash target in this baseline."
- No capture hook: `forward_with_muse_caches` (`src/models/muse_glimmer.rs:52-72`) returns only softcapped logits (`softcap_logits`, `:75-85`, applies `output_multiplier` and `final_logit_softcapping`). No rollback: `MuseCache` (`src/models/muse_glimmer_cache.rs:21-60`) exposes `offset`, `update_and_fetch`, snapshot/restore, but no `trim`. Caches: `MuseCache::Rotating(RotatingKVCache::new(sliding_window))` for sliding layers, `MuseCache::Standard(KVCache::new())` for full layers (`muse_glimmer.rs:100-111`). Attention already builds a causal plus sliding mask when the caller passes `None` (`src/models/muse_glimmer_layers.rs:85-150`), so a multi-token verify forward is causal.
- DFlash drafter model differences: `DFlashConfig` (`src/lib/mlxcel-core/src/drafter/dflash/config.rs:58-129`) has no `layer_types`, `sliding_window`, `rope_parameters`, `draft_window_size`; detection `is_dflash_drafter_config` (`:274-290`) keys on `dflash_config` or `architectures: ["DFlashDraftModel"]`, neither of which a `muse_glimmer_assistant` config has; `DFlashDraftModel::from_weights` (`model.rs:136-180`) reads `fc`, `hidden_norm`, `norm`, `layers.{i}` (no `q_norm`/`k_norm`); `DFlashAttention` (`attention.rs:77-175`) runs SDPA with no mask over a plain `KVCache` (`drafter/dflash/cache.rs`).
- Reusable: `trait Drafter` (`drafter/mod.rs:722-1076`), `trait SpeculativeTarget` and `DFlashGenerator::run` with `speculative_walk` (`drafter/dflash/round_loop.rs:136-270`, `:611-900`, `:407-447`), `drafter_kind_by_model_type` (`drafter/mod.rs:211-235`), the Qwen 3.5 target impl as the template (`src/models/qwen3_5.rs:2480-2560`), rotating rollback primitives `RotatingKVCache::trim` (`src/lib/mlxcel-core/src/cache.rs:5376-5388`) and `enable_speculative_buffer` (`:4346`, used by Gemma 4 MTP with `mtp_rotating_buffer_size = clamp(block * 8, 32, 128)`, `src/models/gemma4_mtp_target.rs:154-157`), dispatch gate `src/server/batch/speculative_burst.rs:1503-1516`.

### Assistant checkpoint

`config.json`:

```
model_type "muse_glimmer_assistant"
hidden_size 6656, intermediate_size 19968, num_hidden_layers 5, num_attention_heads 32,
num_key_value_heads 8, head_dim 128, rms_norm_eps 1e-5, max_position_embeddings 131072,
rope_parameters { rope_theta 500000.0, rope_type "default" },
layer_types ["sliding_attention"] x5, sliding_window 2048,
block_size 16, mask_token_id 201818,
target_layer_ids [1, 13, 25, 37, 49], num_target_layers 52,
vocab_size 202048, bos 200000, eos 200001, pad 200018,
runtime_block_size: optional, draft_window_size: optional
```

(`block_size`, `mask_token_id`, `target_layer_ids`, `runtime_block_size`, `draft_window_size` may also arrive nested under `dflash_config`; flatten them.) Validation: all `layer_types == "sliding_attention"`, `len(target_layer_ids) == num_hidden_layers`, strictly increasing, each `< num_target_layers`. Target validation: `model_type == "muse_glimmer_text"` (the text model inside `muse_glimmer`), `hidden_size == 6656`, 52 layers, `vocab_size == 202048`.

Weight keys (strip a leading `model.`; `.scales`/`.biases` siblings allowed next to a quantized `.weight`), 58 tensors:

```
encoder.fc.weight [6656, 33280]     encoder.output_norm_enc.weight [6656]     norm.weight [6656]
layers.{0..4}.input_layernorm.weight [6656]   layers.{i}.post_attention_layernorm.weight [6656]
layers.{i}.self_attn.q_proj.weight [4096, 6656]   k_proj [1024, 6656]   v_proj [1024, 6656]   o_proj [6656, 4096]
layers.{i}.self_attn.q_norm.weight [128]   k_norm.weight [128]
layers.{i}.mlp.gate_proj.weight [19968, 6656]   up_proj [19968, 6656]   down_proj [6656, 19968]
```

No `embed_tokens`, no `lm_head`. `bind(target)` borrows the target's raw `embed_tokens` lookup (NOT through the target's `embed_norm`) and the target's `lm_head` (NOT followed by `output_multiplier` or `final_logit_softcapping`). Drafter norms are plain RMSNorm with weight `w` (not the Muse centered `1 + w` form).

### Draft step

```
block   = [last_bonus] ++ [mask_token_id] * (bs - 1)          # [1, bs]
h       = embed_tokens_target(block)                            # [1, bs, 6656]
context = output_norm_enc(fc(target_hidden))                    # [1, S, 6656] from [1, S, 5 * 6656]
for layer in layers:
    x = input_layernorm(h)
    if S > sliding_window:  context = context[:, S - sliding_window:]; cache.offset += S - sliding_window; S = sliding_window   # prompt round only
    q = q_norm(q_proj(x))                                       # per-head RMSNorm(head_dim) before RoPE
    kv_in = concat(context, x, axis=1)                          # [1, S + bs, 6656]
    k = k_norm(k_proj(kv_in)); v = v_proj(kv_in)
    q = rope(q, offset = cache.offset + S);  k = rope(k, offset = cache.offset)     # theta 500000, non-interleaved
    K_cache, V_cache = cache.update_and_fetch(k[:, :, :S], v[:, :, :S])            # only the context rows enter the cache
    K = concat(K_cache, k[:, :, S:]); V = concat(V_cache, v[:, :, S:])
    query_start = cache.offset;  key_start = cache.offset - K_cache.len
    mask[i, j] = |(query_start + i) - (key_start + j)| <= sliding_window                # bidirectional, NOT causal
    h = h + o_proj(sdpa(q, K, V, scale = 128^-0.5, mask))
    h = h + mlp(post_attention_layernorm(h))                    # SwiGLU
draft = argmax(lm_head_target(norm(h))[:, 1:])                  # bs - 1 proposals; position 0 (the bonus) is dropped
```

`target_hidden` is the concatenation on the feature axis of the target's post-layer residual streams at `target_layer_ids` (captured after each decoder layer, before the final norm): the whole prompt on the first round, `accepted + 1` rows after a partial accept, all `bs` rows after a full accept. The drafter cache is one `RotatingKVCache(max_size = sliding_window = 2048)` per drafter layer (or a buffered rotating cache of `draft_window_size` when set), append-only: it is never trimmed; each round appends exactly the committed rows. Block size: `min(block_size = 16, runtime_block_size)` or `--draft-block-size`; the first round uses 4 (`dflash_initial_block_size`) and the existing adaptive rule then applies (back off below 30 % / 50 % acceptance, grow by 2 at 85 % acceptance with 75 % full hits).

### Verify rule

`verify_input = [bonus] ++ draft` (`[1, bs]`), one causal target forward capturing the five residual streams and returning `[1, bs, V]` softcapped logits. `target_tokens = argmax` per row; accept the longest matching prefix; emit `draft[..accepted] ++ [target_tokens[accepted]]` (the existing `speculative_walk`). At temperature 0 this is exact greedy decoding and the stream is token-identical to classic decode; at temperature > 0 the DFlash loop keeps its documented argmax-against-argmax behavior (`docs/speculative-acceptance.md`).

### Cache rollback (target)

On a partial accept (`accepted < bs - 1`): `trim = bs - (accepted + 1)`; every target layer cache rewinds by `trim`: `KVCache::trim` for full layers and `RotatingKVCache::trim` for sliding layers (`offset -= n; idx -= n`, no data movement). Because a verify block of `bs` tokens can cross the ring boundary, each sliding target cache is put in buffered speculative mode before the first round with `enable_speculative_buffer(clamp(bs * 8, 32, 128))` (the Gemma 4 MTP rule) so the append-then-rewind never overwrites still-visible window entries. Pinned behavior: prefill 3 tokens, verify 4, rollback `accepted = 1, bs = 4` leaves every cache at `offset == 5`. The drafter cache is not rolled back.

### Surfaces

`mlxcel-server -m <Muse-Glimmer-30B> --model-draft <Muse-Glimmer-30B-assistant> [--draft-block-size N]`, text-only chat requests, B = 1 (B > 1 and multimodal requests decline to classic, as the Qwen 3.5 DFlash path does). `--draft-kind dflash` accepted; auto-detection maps `muse_glimmer_assistant` to `Dflash`. Offline `mlxcel generate` keeps its existing "server-only" rejection for DFlash.

## Implementation plan

1. `src/lib/mlxcel-core/src/drafter/mod.rs:211-235`: insert `("muse_glimmer_assistant", DrafterKind::Dflash)`; in `load_drafter` (`:1097-1133`) route `model_type == "muse_glimmer_assistant"` to a new `dflash::muse::MuseAssistantDrafter::load(path)`.
2. New `src/lib/mlxcel-core/src/drafter/dflash/muse/{config.rs, model.rs, attention.rs}`:
   - `MuseAssistantConfig` (serde, flatten `dflash_config`, fields above, `validate()`), `from_json`.
   - `MuseAssistantAttention`: `q_proj/k_proj/v_proj/o_proj` as `UnifiedLinear`, `q_norm/k_norm` as `RMSNorm(head_dim)`, RoPE via `mlxcel_core::fast_rope(traditional = false, base = rope_theta)`, the split context/proposal projection above, `bidirectional_sliding_mask(query_start, q_len, key_start, k_len, window)` as a boolean `[q_len, k_len]` array (unit-test the 3x6 example: `query_start 10, q_len 3, key_start 7, k_len 6, window 2` gives rows `[F,T,T,T,T,T]`, `[F,F,T,T,T,T]`, `[F,F,F,T,T,T]`), SDPA through `mlxcel_core::scaled_dot_product_attention` with the mask.
   - Decoder layer: reuse `drafter/dflash/mlp.rs` (SwiGLU) and plain `RMSNorm`; `encoder = fc + output_norm_enc`.
   - `MuseAssistantModel { encoder, layers, norm, embed: Option<UnifiedEmbedding>, lm_head: Option<UnifiedLinear> }` with `from_weights` reading the keys above (`sanitize` strips `model.`), `bind(target)` taking `target.embed_tokens_module()` and `target.lm_head_module()` (`LanguageModel` hooks at `src/lib/mlxcel-core/src/generate.rs:456`, `:471`; Muse must return its raw embedding and untied head there), `make_cache` = one `RotatingKVCache::new(sliding_window)` per layer, `draft_block` per the algorithm, `prefer_requested_block_size() == false`, `configured_block_size() == min(16, runtime_block_size)`, `dflash_target_layer_ids()`.
3. `src/models/muse_glimmer.rs`: add `VerifyOutput { logits, hidden_states: Vec<UniquePtr<MlxArray>> }` and `fn forward_speculative(&self, verify_input, caches: &mut [MuseCache], capture_layer_ids) -> VerifyOutput` (same loop as `forward_with_muse_caches` with `mask = None` so the layers build their causal+sliding mask, pushing `h` after layer `i` for `i in capture_layer_ids`); `fn rollback_speculative_cache(&self, caches, accepted, block_size)`; `fn enable_speculative_buffers(&self, caches, buffer)` calling `RotatingKVCache::enable_speculative_buffer` on every `MuseCache::Rotating`. Add `MuseCache::trim(n)` in `src/models/muse_glimmer_cache.rs` dispatching to `KVCache::trim` / `RotatingKVCache::trim`.
4. `impl SpeculativeTarget for MuseGlimmerTextWrapper` (`src/models/muse_glimmer.rs:213-330`): `type Cache = MuseCache`, `type VerifyOut = VerifyOutput`, `verify_forward_with_capture_layers`, `rollback_partial`, `concat_hidden_for_drafter` (`concatenate(hidden_states, -1)` giving `[1, bs, 33280]`), `verify_logits`; sequence-routed variants through `ModelOwnedSequenceState::with_sequence_state(Some(seq_id), ..)`. The VLM wrapper (`src/vision/muse_glimmer_vlm.rs`) forwards these for text-only requests.
5. `src/server/batch/speculative_burst.rs`: add `LoadedModel::MuseGlimmerVLM` to the DFlash target gate (`:1503-1516`) for text-only requests; generalize `Qwen35DFlashTarget` (`:209-229`) into a `DFlashTargetModel` trait implemented by both families (`make_dflash_caches`, `first_hidden_rows` = whole prompt for Muse, `enable_speculative_buffers`), and drive `DFlashGenerator` with `initial_block_size = 4`. Add `initial_block_size` to `DFlashGenerator::new` (`round_loop.rs:317`) if the round loop does not already support a first-round override.
6. `src/server/startup.rs:787-802`: relax the Muse guard to allow `--model-draft` / `--draft-kind dflash` / `--draft-block-size` when the drafter config's `model_type` is `muse_glimmer_assistant`; keep rejecting adapters, KV modes, TP/PP. Update `src/server/muse_glimmer_startup_guard_tests.rs:86-97`.
7. Docs: `docs/supported-models.md` "Speculative decoding" table row `DFlash | Muse Glimmer text requests (B = 1) | mlxcel-server only`; replace the "not a speculative or DFlash target" sentence at `:562`; `docs/speculative-acceptance.md` lists the Muse pairing under the DFlash round loop.

## Validation

(a) Unit tests:
- `src/lib/mlxcel-core/src/drafter/dflash/muse/tests.rs`: `bidirectional_sliding_mask_matches_reference_rows` (the 3x6 example), `config_flattens_dflash_config_and_validates_layer_types`, `draft_block_caches_context_rows_only` (after one draft over a 3-row prompt hidden every drafter cache has `offset == 3`), `prompt_longer_than_window_skips_oldest_context` (S = 2050 with window 2048 leaves `offset == 2050` and 2048 cached rows), `weights_without_embed_or_lm_head_bind_from_target`.
- `src/models/muse_glimmer_tests.rs`: `verify_forward_captures_five_layers_before_final_norm`, `rollback_after_partial_accept_rewinds_every_cache` (prefill 3, verify 4, rollback accepted 1 -> all offsets 5; repeat with a 16-token window and 40 prefilled tokens so the sliding caches are wrapped, using the speculative buffer), `speculative_block_logits_match_single_token_decode` on a synthetic 2-layer config.
- `src/server/muse_glimmer_startup_guard_tests.rs`: `muse_glimmer_accepts_assistant_drafter` and `muse_glimmer_rejects_non_assistant_drafter`.
- Round loop: `dflash_round_loop_initial_block_size_four` in `round_loop.rs` tests.

(b) Real checkpoints: `meta-models/Muse-Glimmer-30B` (or `mlx-community/Muse-Glimmer-30B-4bit` for the target) with `meta-models/Muse-Glimmer-30B-assistant` (bf16; quantize locally with the checkpoint's own conversion tool if needed):

```
./target/release/mlxcel-server -m models/Muse-Glimmer-30B-4bit --model-draft models/Muse-Glimmer-30B-assistant --port 8080
curl -s localhost:8080/v1/chat/completions -d '{"model":"m","messages":[{"role":"user","content":"Write a 200-word summary of how tides work."}],"max_tokens":256,"temperature":0}'
```

Acceptance: the 256 token ids are identical to the same server without `--model-draft`; `timings` report mean accepted length above 1 and tok/s above classic; a request with an image declines to classic with one log line; a conversation longer than 2048 tokens (sliding caches wrapped) still matches classic output token for token.

## Acceptance criteria

- [x] `muse_glimmer_assistant` is detected, loaded with the key layout above, bound to the Muse target's raw embedding and untied head, and rejected on any mismatched target.
- [x] The Muse target captures the five residual streams, verifies a block causally, and rolls back full and sliding caches (buffered) to the committed prefix.
- [x] Temperature 0 output is token-identical to classic decode, including past the 2048-token window.
- [x] `mlxcel-server` accepts the pairing while still rejecting adapters, KV modes and TP/PP for Muse Glimmer.
- [x] docs/supported-models.md updated
- [ ] cargo test --workspace --profile test-fast --features metal,accelerate passes
- [ ] cargo clippy --workspace --all-targets -- -D warnings and cargo fmt --all -- --check pass

## Out of scope

- Batched (B > 1) Muse DFlash and multimodal (image) requests under the drafter.
- Offline `mlxcel generate` DFlash construction (server-only, as for Qwen 3.5).
- `draft_window_size` buffered-rotating drafter caches (accept the config key, use the plain rotating window).
